### DISCUSSION TITLE
We are currently starting to look for WebGPU bindings to work in a browser.

### DISCUSSION THREAD

**Discussion by BradChase2011** [amerkoleci/Alimer.Bindings.WebGPU#6] [General]
We are just starting research into what bindings to go with for a browser version of our engine which currently has endpoints for DirectX and Vulkan.  We use the Vortice bindings for the DirectX side (and love them).  That said, is this repository expected or can be used in a browser under WASM?  If so, are there any examples we could take a peak at and try to run locally?  Anything like a simple triangle would be perfect.  If not, is anyone aware of bindings and samples that work under wasm in the browser?

Thanks much!
Brad

**Comment by BradChase2011** [comment]
I went ahead and created an example using another route to take to interface with C#, WebGPU in WASM (Blazor).  Is there any way to move Alimer.Bindings.WebGPU over to wasm?  Are there current road blocks?

https://github.com/BradChase2011/CSharpBlazorWebGPUExample

Thanks!
Brad

**Reply by amerkoleci** [reply]
Hi,
Sorry for not replying quickly, was a little bit busy.

I never tried something using C# and targeting browser/wasm thus is impossible for me to give advice and help here.

When possible I'll try to investigate web C# stuff and update you, taking look at your repo as well.

Cheers!
Amer

**Reply by BradChase2011** [reply]
No worries at all, its just a simple example drawing a triangle in the browser using webgpu in C# based on two other repositories.  I couldn't find a single example or repository anywhere that had a simple way to use WebGPU (browser) in C# that worked with the tools existing today.  It would be really nice to have your bindings working because they line up much closer to the Vulkan and DirectX 12 bindings rather than straight interop calls used in the JS/C/CPP that my example uses.

My hope is there would be a quick way since the headers in theory should line up well but then again i am pretty naive on WebGPU at this point.
